### PR TITLE
[render preview] Add Delete Comment Functionality

### PR DESCRIPTION
**Pull request by Saurabhchaudhary9799**
### Description:
This PR implements the ability for users to delete comments on recipes. The following changes were made:
- Added a new handleDeleteComment function in the OneRecipe component.
- Integrated an API call to delete comments based on their ID.
- Updated the frontend to reflect the removal of a comment after it has been deleted.
- Ensured that only authorized users can delete comments by including a token in the request headers.

### Type of change 
-feature added

### Screen Recording : 
[Screencast from 2024-10-19 11-55-25.webm](https://github.com/user-attachments/assets/57815215-b4cf-4417-ac0e-a7f77e6397b3)

### CODE REVIEW

**Comment by Saurabhchaudhary9799** [comment]
@AlfiyaSiddique , Please check my PR and add level3 label on this PR

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 please update your PR

File: backend/Controllers/RecipeController.js [hunk 1]
```diff
@@ -363,16 +363,39 @@ const getComments = async (req, res) => {
...
     res.status(500).json({ success: false, message: "Internal server error" });
   }
 };
 
+const deleteComment = async (req,res) => {
+  try {
+    // Extract comment ID from the request parameters
+    const { commentId } = req.params;
+// console.log(commentId)
+    // Find the comment by ID and delete it
```
Comment (AlfiyaSiddique) [inline backend/Controllers/RecipeController.js:379]: remove this console.log comment

File: frontend/src/Pages/OneRecipe.jsx [hunk 2]
```diff
@@ -63,6 +67,25 @@ const OneRecipe = () => {
...
+    const handleDeleteComment = async (commentId) => {
+      try {
+        // Send a DELETE request to the backend to remove the comment by ID
+        const response = await axios.delete(`${backendURL}/api/recipe/deletecomment/${commentId}`, {
+          headers: {
+            Authorization: `Bearer ${token}`, // Use token for authentication
+          },
+        });
+    console.log(response)
+        // Remove the deleted comment from the comments array in the frontend
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/OneRecipe.jsx:79]: remove this console.log

File: package.json [hunk 3]
```diff
@@ -1,5 +0,0 @@
-{
-  "devDependencies": {
-    "nodemon": "^3.1.7"
-  }
-}
```
Comment (AlfiyaSiddique) [inline package.json:5]: @Saurabhchaudhary9799 why did you delete this?

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 Some comments on your code, please fix them
also remove the tanddor.png image we store the image of the recipe in realtime so no need of static image

**Comment by Saurabhchaudhary9799** [comment]
@AlfiyaSiddique , i will fix the changes

File: package.json (hunk 3 above)
Comment (Saurabhchaudhary9799) [reply package.json:5]: @AlfiyaSiddique , I installed some node_modules in TastyTrials directory by mistake , so i deleted this. This package.json file  is not necessary.

**Comment by Saurabhchaudhary9799** [comment]
> @Saurabhchaudhary9799 Some comments on your code, please fix them also remove the tanddor.png image we store the image of the recipe in realtime so no need of static image

@AlfiyaSiddique , I fixed the code. Please check PR once again.

**Comment by Saurabhchaudhary9799** [comment]
> > @Saurabhchaudhary9799 Some comments on your code, please fix them also remove the tanddor.png image we store the image of the recipe in realtime so no need of static image
> 
> @AlfiyaSiddique , I fixed the code. Please check PR once again.

@AlfiyaSiddique , If anything is  wrong in the code  changes , tell me.

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 Please completely remove the images, package-lock.json, and package.json file 
image
